numba/parfor.py

```diff
@@ -1915,3 +2110,11 @@ def parfor_typeinfer(parfor, typeinferer):
 
 
 typeinfer.typeinfer_extensions[Parfor] = parfor_typeinfer
+
+@infer_global(reduce)
+class ReduceInfer(AbstractTemplate):
+    def generic(self, args, kws):
+        assert not kws
+        assert len(args) == 3
+        assert isinstance(args[1], types.Array)
```

Comment:
This restriction should be noted in the documentation, along with requiring an initial value.

Reply:
Done.